c7n/resources/fsx.py

```diff
@@ -0,0 +1,159 @@
+        return resources
+
+
+@FSx.action_registry.register('mark-for-op')
+class MarkForOpFileSystem(TagDelayedAction):
+    concurrency = 2
+    batch_size = 5
+    permissions = ('fsx:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.tag_resource(ResourceARN=r['ResourceARN'], Tags=tags)
+
+
+@FSx.action_registry.register('tag')
+class TagFileSystem(Tag):
+    concurrency = 2
+    batch_size = 5
+    permissions = ('fsx:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.tag_resource(ResourceARN=r['ResourceARN'], Tags=tags)
+
+
+@FSx.action_registry.register('remove-tag')
+class UnTagFileSystem(RemoveTag):
+    concurrency = 2
+    batch_size = 5
+    permissions = ('fsx:UntagResource',)
+
+    def process_resource_set(self, resources, tag_keys):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.untag_resource(ResourceARN=r['ResourceARN'], TagKeys=tag_keys)
+
+
+@FSx.action_registry.register('update')
+class UpdateFileSystem(BaseAction):
+    """
+    Update FSx resource configurations
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+            - name: update-fsx-resource
+              resource: fsx
+              actions:
+                - type: update
+                  WindowsConfiguration:
+                    AutomaticBackupRetentionDays: 1
+                    DailyAutomaticBackupStartTime: '04:30'
+                    WeeklyMaintenanceStartTime: '04:30'
+                  LustreConfiguration:
+                    WeeklyMaintenanceStartTime: '04:30'
+
+    Reference: https://docs.aws.amazon.com/fsx/latest/APIReference/API_UpdateFileSystem.html
+    """
+    permissions = ('fsx:UpdateFileSystem',)
+
+    schema = type_schema(
+        'update',
+        WindowsConfiguration={'type': 'object'},
+        LustreConfiguration={'type': 'object'}
+    )
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.update_file_system(
+                FileSystemId=r['FileSystemId'],
+                WindowsConfiguration=self.data.get('WindowsConfiguration', {}),
+                LustreConfiguration=self.data.get('LustreConfiguration', {})
+            )
+
+
+@FSx.action_registry.register('backup')
+class BackupFileSystem(BaseAction):
+    """
+    Create Backups of File Systems
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+            - name: backup-fsx-resource
+              resource: fsx
+              actions:
+                - type: backup
+                  tags:
+                    some-key: some-value
```

Comment:
we should probably have an option here to copy the tags from the fs that's being backed up as well.

Reply:
so we seem to have two different 'copy-tags' implementations, one is just a boolean to copy the tags from the source to the destination and the other enumerates the tags to be copied.

e.g.:

```yaml
policies:
   - name: ec2-nightly-backup
     resource: ec2
     actions:
       - type: snapshot
         copy-tags:
           - Owner
```

or

```yaml
policies:
  - name: rds-snapshot-retention
     resource: rds
     actions:
      - type: retention
         days: 7
         copy-tags: true
         enforce: exact
```

Reply:
thanks for checking on the consistency aspect, still noodling, I could see a list with a special value for `all` or just retaining the boolean, to keep some semblance of consistency. its a judgement call, but I think just retaining the boolean seems like the better option.

Reply:
I can see use cases on both sides, re copy en block and explicit names, copy en block seems common, but wdyt?

Reply:
i think standardizing on the list makes the most sense from a customizability standpoint, especially since the Name tag is used as the backup's name and if we were to copy all tags over from the fs the backup name would just be equal to the fs name.

maybe something like so to copy all tags to prevent stepping on existing tags (which maybe named all):

```yaml
actions:
  - type: tag
    copy-tags:
      - C7N_ALL_TAGS
```

or to copy some tags:
```yaml
actions:
  - type: tag
    copy-tags:
      - Application
      - Owner
```

Reply:
yeah.. its more customization but i'm not clear if there is good use case to force users to do explicit config here, when the common case is just copy them all. its also a magic value thats idiosyncratic to a particular action, if the common use case is actually just to copy as many as possible, ie whats the use case to only partially. previously it was on the basis of tag limit, but thats not really been an issue since the raise to 50 tags per resource a while ago.

Reply:
ok, sounds good